### PR TITLE
python3Packages.tensorflow-bin_2: 2.2.0 -> 2.4.0

### PR DESCRIPTION
**Pull request by nbren12**
###### Motivation for this change

Update `tensorflow-bin_2` version to match the source build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nbren12** [comment]
@alexvorobiev you might be interested in this.

**Comment by nbren12** [comment]
Bumping this issue @jonringer.

Would it make sense to merge before splitting off the new release? This makes the binary and source builds of tensorflow have a consistent version.

File: pkgs/development/python-modules/tensorflow/binary-hashes.nix [hunk 1]
```diff
@@ -1,51 +1,39 @@
...
-  url = "https://storage.googleapis.com/tensorflow/mac/cpu/tensorflow-2.1.0-cp37-cp37m-macosx_10_9_x86_64.whl";
-  sha256 = "1hh4n0d97mrq35cmmsrnlmcv9vlswsyjy368lj3pda3y9dvck3rf";
+  url = "https://storage.googleapis.com/tensorflow/mac/cpu/tensorflow-2.4.0-cp37-cp37m-macosx_10_14_x86_64.whl";
+  sha256 = "1mgfyjgcwvx5jzawrpfnbch5sqw7kpnzp35rfxj22kwdsl28r47r";
+};
+mac_py_38_cpu = {
+  url = "https://storage.googleapis.com/tensorflow/mac/cpu/tensorflow-2.4.0-cp38-cp38-macosx_10_14_x86_64.whl";
+  sha256 = "034qh0nk786wxzma58179g086x479c3pd9vi8v8p26grs6f2fm0p";
 };
 }
```
Comment (jonringer) [inline pkgs/development/python-modules/tensorflow/binary-hashes.nix:39]: needs another:
```
attribute 'mac_py_39_cpu' missing, at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-7/pkgs/development/python-modules/tensorflow/bin.nix:67:15
```

or disable for python >= 3.9 
```
disabled = pythonAtLeast "3.9";
```

Comment (nbren12) [reply pkgs/development/python-modules/tensorflow/binary-hashes.nix:39]: Thanks! There are no 3.9 binary wheels in google's cloud bucket, so I did the latter.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 121998 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>python38Packages.tensorflow-bin (python38Packages.tensorflow-bin_2)</li>  </ul></details>

**Pull request merged**